lib/apiservers/engine/backends/system.go

```diff
@@ -381,3 +408,45 @@ func FetchVolumeStores(client *client.PortLayer) (string, error) {
 
 	return strings.Join(res.Payload.Stores, " "), nil
 }
+
+// RegistrySetContains checks if a URL is within the set of insecure registries
+func RegistrySetContains(registrySet []url.URL, remoteURL *url.URL) bool {
+	containedOK := false
+	remoteIP := net.ParseIP(remoteURL.Hostname())
+	remoteDomains, _ := net.LookupAddr(remoteURL.Hostname())
+
+	for _, rs := range registrySet {
+		// if CIDR was provided for insecure registry range, check if login URL within that ip net
+		if strings.Index(rs.Path, "/") != -1 {
```

Comment:
Instead of the three `if` clauses, can I suggest that you use an interface instead? Concrete implementations for the three cases can then have the logic in them, and you just make one call to the method here, e.g. `Contains`, etc.